java/squeek/appleskin/mixin/GuiMixin.java

```diff
@@ -36,6 +37,10 @@ private void renderTooltipCaptureLocals(MatrixStack matrixStack, List tooltip, i
 	@Inject(at = @At(value = "INVOKE", target = "Lnet/minecraft/client/util/math/MatrixStack;pop()V", ordinal = 0), method = "renderOrderedTooltip(Lnet/minecraft/client/util/math/MatrixStack;Ljava/util/List;II)V")
 	private void renderTooltip(MatrixStack matrixStack, List tooltip, int mouseX, int mouseY, CallbackInfo info)
 	{
+		// all mixins should not take effect until ModConfig initialized.
+		if (ModConfig.INSTANCE == null)
+			return;
```

Comment:
I don't really like the idea of this being in the mixin classes themselves. I think these checks would be better in the `TooltipOverlayHandler`/`HUDOverlayHandler` functions that the mixins call.

It might not be not an issue here, but it can be dangerous to import/load classes that import/load external classes (in this case, ModConfig imports/loads the AutoConfig/Cloth stuff) at class loading-time, since it can potentially screw with other class loaders (like loading a class before another mod's mixin stuff is initialized so the other mod never gets a chance to apply their mixin). Note that Mixins might solve this class loader ordering problem, I'm mostly going off how it worked in Forge without Mixin. I think it's still better to be safe here and keep the mixin classes themselves as simple as possible.